src/index/DeltaTriples.cpp

```diff
@@ -237,31 +240,37 @@ ReturnType DeltaTriplesManager::modify(
         };
         auto writeAndUpdateSnapshot = [&updateSnapshot, &deltaTriples,
                                        writeToDiskAfterRequest]() {
+          ad_utility::Timer timer{ad_utility::Timer::InitialStatus::Started};
+          DeltaTriplesModifyTimings timings;
           if (writeToDiskAfterRequest) {
-            // TODO<RobinTF> Find a good way to track the time it takes to write
-            // this and store it into the corresponding metadata object.
             deltaTriples.writeToDisk();
+            timings.diskWritebackTime_ = timer.msecs();
+            timer.start();
           }
           updateSnapshot();
+          timings.snapshotUpdateTime_ = timer.msecs();
+          return timings;
         };
 
         if constexpr (std::is_void_v<ReturnType>) {
           function(deltaTriples);
-          writeAndUpdateSnapshot();
+          return writeAndUpdateSnapshot();
         } else {
           ReturnType returnValue = function(deltaTriples);
-          writeAndUpdateSnapshot();
-          return returnValue;
+          auto timings = writeAndUpdateSnapshot();
+          return std::make_tuple(returnValue, timings);
```

Comment:
```suggestion
          return {std::move(returnValue), std::move(timings)};
```

Reply:
It seems that we cannot get rid of the `make_tuple` because the return type is not directly `tuple` and the compiler does not detect that it is always `tuple` here.